File: src/main/java/com/example/ags/query/HospitalView.java
package com.example.ags.query;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.Id;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class HospitalView {
    @Id
    private String hospCode;

    @ElementCollection
    private List<String> wards;
}

File: src/main/java/com/example/ags/query/HospitalView.kt
package com.example.ags.query

import org.springframework.data.jpa.repository.JpaRepository
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class HospitalView(
        @Id val hospCode: String
)

interface HospitalViewRepository : JpaRepository<HospitalView, String>